### Домашняя работа:Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

In [15]:
import numpy as np
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def bernulli(p, n, k):
    q=1-p
    return combinations(n, k) * p**k * q**(n-k)
    

#### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

В данном случае случайная величина попадания является дискретной, следовательно, можем применить формулу Бернулли:

In [20]:
p=0.8
n=100
k=85
bernulli(p,n,k)

0.048061793700746355

#### Ответ: 0.048061793700746355

#### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек

#### а). Какова вероятность, что ни одна из них не перегорит в первый день?

In [16]:
p=0.0004
n=5000
k=0
bernulli(p,n,k)

0.13528114551440706

#### Ответ: 0.13528114551440706

#### б). Какова вероятность, что перегорят ровно две?

In [17]:
p=0.0004
n=5000
k=2
bernulli(p,n,k)

0.2707247150266753

#### Ответ: 0.2707247150266753

#### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

если монета не фальшивая, и не падает на ребро (что у меня было несколько раз в жизни), то вероятность выпадения орла и решки одинаковая: 0.5

In [21]:
p = 0.5
n = 144
k = 70
bernulli(p,n,k)

0.06281178035144776

#### Ответ: 0.06281178035144776

#### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.

#### a). Какова вероятность того, что все мячи белые?

In [26]:
#вероятность для 1 ящика:
PW1 = combinations(7,2)/combinations(10,2)
#вероятность для 2 ящика:
PW2 = combinations(9,2)/combinations(11,2)
#общая вероятность:
PW1*PW2

0.3054545454545455

#### Ответ: 0.3054545454545455

#### б). Какова вероятность того, что ровно два мяча белые?

Варианты:
#### 1 ящик - 2 белых мяча, 2 ящик - 0 белых мячей
#### 1 ящик - 1 белый мяч, 2 ящик - 1 белый мяч
#### 1 ящик - 0 белых мячей, 2 ящик - 2 белых мяча

In [29]:
# 1 вариант
P11 = combinations(7,2) * combinations(3,0) / combinations(10,2)
P12 = combinations(9,0) * combinations(2,2) / combinations(11,2)

# 2 вариант
P21 = combinations(7,1) * combinations(3,1) / combinations(10,2)
P22 = combinations(9,1) * combinations(2,1) / combinations(11,2)

# 3 вариант
P31 = combinations(7,0) * combinations(3,2) / combinations(10,2)
P32 = combinations(9,2) * combinations(2,0) / combinations(11,2)

# общая вероятность:
P11*P12 + P21*P22 + P31*P32

0.20484848484848486

#### Ответ: 0.20484848484848486

#### в). Какова вероятность того, что хотя бы один мяч белый?

вероятность того, что хотя бы один мяч белый равна 1 минус вероятность того, что все мячи чёрные

In [31]:
# вероятность, что все мячи чёрные:

#вероятность для 1 ящика:
PW1 = combinations(3,2)/combinations(10,2)
#вероятность для 2 ящика:
PW2 = combinations(2,2)/combinations(11,2)
#вероятность, что хотя бы один мяч белый:
1 - PW1*PW2

0.9987878787878788

#### Ответ: 0.9987878787878788